.buildkite/scripts/pipelines/security_solution_quality_gate/api-integration-tests.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+if [ -z "$1" ]
+  then
+    echo "No target script from the package.json file, is supplied"
+    exit 1
+fi
+
+source .buildkite/scripts/common/util.sh
+.buildkite/scripts/bootstrap.sh
+
+buildkite-agent meta-data set "${BUILDKITE_JOB_ID}_is_test_execution_step" "true"
+
+echo "--- Serverless Security Second Quality Gate"
+cd x-pack/test/security_solution_api_integration
+set +e
+
+QA_API_KEY=$(retry 5 5 vault read -field=qa_api_key secret/kibana-issues/dev/security-solution-qg-enc-key)
+
+# Generate a random 5-digit number
+random_number=$((10000 + $RANDOM % 90000))
+ENVIRONMENT_DETAILS=$(curl --location 'https://global.qa.cld.elstc.co/api/v1/serverless/projects/security' \
```

Comment:
How do we know what commit these are running against?  Can we assume the image promoted to QA is the correct test environment?

Reply:
Thats exactly the logic behind this quality gate.

Reply:
Can you help me understand the workflow?  This is running on QA, so this quality gate is expected to run only before promotion to staging?

Reply:
Thanks, understood.